squash-honeycomb.ipynb

```diff
@@ -0,0 +1,381 @@
+    "Actually, just the scalar metrics and some context information."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Create the `squash-demo` dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "libhoney.init(writekey=HONEY_API_KEY, dataset=\"squash-demo\")\n",
+    "builder = libhoney.Builder()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And loop over all jobs:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from datetime import datetime\n",
+    "    \n",
+    "jobs = requests.get(SQUASH_API_URL + \"/jobs\").json()\n",
+    "\n",
+    "for job_id in jobs['ids']:\n",
+    "\n",
+    "    r = requests.get(SQUASH_API_URL + \"/job/{}\".format(job_id)).json()\n",
+    "    \n",
+    "    if r['ci_dataset'] == 'unknown' or r['ci_dataset'] == 'decam':\n",
+    "        continue\n",
+    "    \n",
+    "    print('Sending event for job {}...'.format(job_id))\n",
+    "\n",
+    "    \n",
+    "    # Spawn a new event and override the timestamp\n",
+    "    event = builder.new_event()\n",
+    "    event.add_field('id', job_id)\n",
+    "    event.add_field('filter_name', r['meta']['filter_name'])\n",
+    "    event.add_field('dataset', r['ci_dataset'])\n",
+    "    \n",
+    "    for meas in r['measurements']:\n",
+    "        event.add_field(meas['metric'], meas['value'])\n",
+    "    \n",
+    "    event.created_at = datetime.strptime(r['date_created'], \"%Y-%m-%dT%H:%M:%SZ\")\n",
+    "    \n",
+    "    event.send()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Querying the dataset\n",
+    "\n",
+    "Querying a Honeycomb dataset in a particular way will produce a series of events (or a time series). The query builder is great tool for fast exploration from the Honeycomb UI. It is also possible to [specify queries programatically](https://docs.honeycomb.io/api/query-specification/)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The SQuaSH `validate_drp.AM1` metric on a given dataset and filter can be obtained obtained from this query:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "query = {\n",
+    "    \"breakdowns\": [\n",
+    "        \"dataset\", \"filter_name\"\n",
+    "    ],\n",
+    "    \"calculations\": [\n",
+    "        {\"column\": \"validate_drp.AM1\", \"op\": \"AVG\"}\n",
+    "    ],\n",
+    "    \"filters\":[\n",
+    "      {\"column\": \"dataset\", \"op\": \"=\", \"value\": \"validation_data_hsc\"},\n",
+    "      {\"column\": \"filter_name\", \"op\": \"=\", \"value\": \"HSC-R\"}\n",
+    "    ],\n",
+    "    \"filter_combination\": \"AND\"\n",
```

Comment:
I realize I should read the docs, but this implies that all filters must be either OR'd or AND'd.  Is that really true?

Reply:
filter_combination: either "AND" or "OR". If multiple filters are specified, filter_combination determines how they are applied; set to "OR" to match ANY filter in the filter list. Defaults to "AND".

Reply:
OK, unless I'm missing something, this can be pretty restrictive.  I.e. I can't say something like (filter_name is 'r' or filter_name is 'i') and date_obs > yesterday.

Reply:
Good point. I guess it simplifies implementation, but I agree it is pretty restrictive. I will ask Honeycomb support about that.